### PR TITLE
SPCTR-2500 - Fix: Added Box Shadow Generation Helpers

### PR DESCRIPTION
**Pull request by bsf-zanev**
### Description
Added JS and PHP Box Shadow Helpers, and implemented in Countdown block with JS Fallback to avoid creating Backward Compatibility.
Also added a Separate Hover Box Shadows Toggle ( Default True ) to easily set shadows without hover.

### Types of changes
Bug fix (non-breaking change which fixes an issue)

### How has this been tested?
Tested using the Countdown Block in the editor and frontend with and without normal ( and hover ) box shadows.

### Checklist:
- [x] My code is tested
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: classes/class-uagb-block-helper.php [hunk 1]
```diff
@@ -1831,5 +1831,60 @@ public static function get_precise_percentage( $divisions ) {
...
+				return '';
+			}
+
+			// Get the CSS units for the optional shadow properties.
+			$blur   = UAGB_Helper::get_css_value( $blur, $blur_unit );
+			$spread = UAGB_Helper::get_css_value( $spread, $spread_unit );
+
+			// Return the full CSS if blur is set, else return the required css.
+			return $blur ? (
+				( $horizontal ? $horizontal : 0 ) . ' ' . ( $vertical ? $vertical : 0 ) . ' ' . $blur . ( $spread ? " {$spread}" : '' ) . ' ' . ( $color ? $color : $alt_color ) . ( 'outset' === $position ? '' : " {$position}" )
```
[the remark this answers is not in the document]
Comment (bsf-zanev) [reply classes/class-uagb-block-helper.php]: I've added the same, the CSS generation now works in one line.

File: blocks-config/uagb-controls/generateShadowCSS.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+	// Get the CSS units for the optional shadow properties.
+	blur = generateCSSUnit( blur, blurUnit );
+	spread = generateCSSUnit( spread, spreadUnit );
+
+	// Return the full CSS if blur is set, else return the required css.
+	return blur ? (
+		`${ horizontal ? horizontal : 0 } ${ vertical ? vertical : 0 } ${ blur }${ spread ? ` ${ spread }` : '' } ${ color ? color : altColor }${ ( 'outset' === position ) ? '' : ` ${ position }` }`
+	) : (
+		`${ horizontal ? horizontal : 0 } ${ vertical ? vertical : 0 } ${ color ? color : altColor }${ ( 'outset' === position ) ? '' : ` ${ position }` }`
```
[the remark this answers is not in the document]
Comment (bsf-zanev) [reply blocks-config/uagb-controls/generateShadowCSS.js]: I've added the same, the CSS generation now works in one line.

File: classes/class-uagb-block-helper.php [hunk 3]
```diff
@@ -1831,5 +1831,60 @@ public static function get_precise_percentage( $divisions ) {
...
+			if ( ! $color && ! $alt_color ) {
+				return '';
+			}
+
+			// Get the CSS units for the required shadow properties.
+			$horizontal = UAGB_Helper::get_css_value( $horizontal, $horizontal_unit );
+			$vertical   = UAGB_Helper::get_css_value( $vertical, $vertical_unit );
+
+			// If both required properties don't exist, shaow won't be generated.
+			if ( ! $horizontal && ! $vertical ) {
```
[the remark this answers is not in the document]
Comment (bsf-zanev) [reply classes/class-uagb-block-helper.php]: I've updated the logic to generate CSS with 0 0 if needed, so now users can easily set blur ( and spread for box shadow )

File: classes/class-uagb-block-helper.php [hunk 4]
```diff
@@ -1831,5 +1831,60 @@ public static function get_precise_percentage( $divisions ) {
...
+		 * ( 'alt_color' ) should be set as the attribute used for ( 'color' ) in Box/Text Shadow Normal CSS.
+		 *
+		 * @param array $shadow_properties  Array containing the necessary shadow properties.
+		 * @return string                   The generated border CSS or an empty string on early return.
+		 *
+		 * @since x.x.x
+		 */
+		public static function generate_shadow_css( $shadow_properties ) {
+			// Get the Object Properties.
+			$horizontal      = isset( $shadow_properties['horizontal'] ) ? $shadow_properties['horizontal'] : '';
```
[the remark this answers is not in the document]
Comment (bsf-zanev) [reply classes/class-uagb-block-helper.php:1852]: I've updated the logic to generate CSS with 0 0 if needed, so now users can easily set blur ( and spread for box shadow )

**Review by adi3890 — APPROVED** [review]

**Review by adi3890 — APPROVED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**